pymc3/distributions/discrete.py

```diff
@@ -359,10 +359,7 @@ class Categorical(Discrete):
     """
     def __init__(self, p, *args, **kwargs):
         super(Categorical, self).__init__(*args, **kwargs)
-        try:
-            self.k = np.shape(p)[-1].tag.test_value
-        except AttributeError:
-            self.k = np.shape(p)[-1]
+        self.k = np.shape(p)[-1]
```

Comment:
Yes, we need to leave this in, otherwise `k` ends up being a tensor. It needs to be a scalar.